server/src/main/java/org/elasticsearch/common/bytes/PagedBytesReference.java

```diff
@@ -48,6 +49,29 @@ public BytesReference slice(int from, int length) {
         return new PagedBytesReference(byteArray, offset + from, length);
     }
 
+    @Override
+    public BytesReference copy(int from, int length) {
```

Comment:
I am not very familiar with the details behind the bytes reference, so I am not sure what this method is trying to do in summary?

Reply:
I think this should explain it: https://github.com/elastic/elasticsearch/pull/100615

Reply:
Ah that is long :) I did not read it, but I think I got the rough idea from reading `iterator()` that we want to read whole pages so that they are not materialized.